src/coreclr/vm/ceeload.cpp

```diff
@@ -4829,28 +4827,28 @@ VASigCookie *Module::GetVASigCookieWorker(Module* pDefiningModule, Module* pLoad
             pCookie->pNDirectILStub = NULL;
             pCookie->sizeOfArgs = sizeOfArgs;
             pCookie->signature = vaSignature;
-            pCookie->classInstCount = classInstCount;
-            pCookie->methodInstCount = methodInstCount;
             pCookie->pLoaderModule = pLoaderModule;
 
             AllocMemTracker amt;
         
             if (classInstCount != 0)
             {
-                pCookie->classInst = (TypeHandle*)(void*)amt.Track(pLoaderAllocator->GetHighFrequencyHeap()->AllocMem(S_SIZE_T(classInstCount * sizeof(TypeHandle))));
+                TypeHandle* pClassInst = (TypeHandle*)(void*)amt.Track(pLoaderAllocator->GetHighFrequencyHeap()->AllocMem(S_SIZE_T(classInstCount * sizeof(TypeHandle))));
```

Comment:
```suggestion
                TypeHandle* pClassInst = (TypeHandle*)(void*)amt.Track(pLoaderAllocator->GetHighFrequencyHeap()->AllocMem(S_SIZE_T(classInstCount) * S_SIZE_T(sizeof(TypeHandle))));
```
Casts to `S_SIZE_T` are meant to be done on the individual operands to catch potential integer overflow.